src/main/groovy/edu/oregonstate/mist/catalogapitest/resources/InstructorResource.groovy

```diff
@@ -0,0 +1,210 @@
+     *
+     * @param cid
+     * @return response containing the result or error message
+     */
+    @GET
+    @Path('{cid}')
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getByCid(@PathParam('cid') IntParam cid) {
+        Instructor instructors = instructorDAO.getByCid(cid.get())
+
+        Response returnResponse
+
+        if (instructors == null) {
+            ErrorPOJO returnError = new ErrorPOJO("Resource Not Found.", Response.Status.NOT_FOUND.getStatusCode())
+            returnResponse = Response.status(Response.Status.NOT_FOUND).entity(returnError).build()
+        } else {
+            returnResponse = Response.ok(instructors).build()
+        }
+
+        return returnResponse
+    }
+
+    /**
+     * Responds to PUT requests depending on the state of the instructor object,
+     * either through updating existing object with PUT request or creating with
+     * POST request if not already existing.
+     *
+     * @param cid
+     * @param newInstructor
+     * @return response containing the result or error message
+     */
+    @PUT
+    @Path('{cid}')
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response putByCid(@PathParam("cid") Integer cid, Instructor newInstructor) {
+
+        Response returnResponse
+
+        Instructor checkForInstructorCid = instructorDAO.getByCid(cid)
+
+        // If instructor does not already exist - POST it!
+        if (checkForInstructorCid == null) {
+            instructorDAO.postByCid(cid, newInstructor.getFirstInitial(), newInstructor.getLastName(), newInstructor.getNumberOfCourses())
+            returnResponse = Response.created().build()
+        } else {
+
+            // Otherwise PUT it!
+            Optional<String> newFirstInitial = Optional.of( newInstructor.getFirstInitial() )
+            Optional<String> newLastName = Optional.of( newInstructor.getLastName() )
+            Optional<Integer> newNumberOfCourses = Optional.of( newInstructor.getNumberOfCourses() )
+
+            instructorDAO.putByCid(cid, newFirstInitial.or(checkForInstructorCid.getFirstInitial()),
+                                        newLastName.or(checkForInstructorCid.getLastName()),
+                                        newNumberOfCourses.or(checkForInstructorCid.getNumberOfCourses())
+            )
+
+            returnResponse = Response.ok().build()
+        }
+
+        return returnResponse
+    }
+
+    /**
+     * Responds to DELETE requests and removes an instructor object.
+     *
+     * @param cid
+     * @return response containing the result or error messages
+     */
+    @DELETE
+    @Path('{cid}')
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response deleteByCid(@PathParam('cid') Integer cid) {
+
+        //TODO add authentication for this method
+        instructorDAO.deleteByCid(cid)
+        Response returnResponse = Response.ok().build()
+
+        return returnResponse
```

Comment:
`returnResponse` is returned immediately after it is defined, so it is not necessary to allocate a variable (unless you wish to make code self-documenting)